src/test/java/net/minecraftforge/fml/test/BetterModDependenciesTest.java

```diff
@@ -0,0 +1,55 @@
+package net.minecraftforge.fml.test;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+
+import net.minecraftforge.fml.common.Loader;
+import net.minecraftforge.fml.common.LoaderException;
+import net.minecraftforge.fml.common.versioning.ArtifactVersion;
+
+public class BetterModDependenciesTest
+{
+
+    @Test
+    public void test()
+    {
+        Loader.computeDependencies(null, null, null, null);
+
+        Loader.computeDependencies("", null, null, null);
+
+        {
+            Set<ArtifactVersion> requirements = new HashSet<ArtifactVersion>();
+            Loader.computeDependencies("required:supermod3000[1.2,)", requirements, null, null);
+            assertFalse(requirements.isEmpty());
```

Comment:
This is poor testing, you should check that the length is 1 and the values in the first ArtifactVersion match what you expect.